<!DOCTYPE html>
<html>
  <script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-41619785-1', 'auto');
  ga('send', 'pageview');

</script>

  <head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  <title>Haplotypewriter</title>
  <meta name="description" content="A tangled bank of biology, computers, and Star Wars. More at aaronhardin.com
">

  <link rel="stylesheet" href="/css/main.css">
  <link rel="canonical" href="http://haplotypewriter.com/">
  <link rel="alternate" type="application/rss+xml" title="Haplotypewriter" href="http://haplotypewriter.com/feed.xml" />
</head>


  <body>

    <header class="site-header">

  <div class="wrapper">

    <a class="site-title" href="/">Haplotypewriter</a>

    <nav class="site-nav">
      <a href="#" class="menu-icon">
        <svg viewBox="0 0 18 15">
          <path fill="#424242" d="M18,1.484c0,0.82-0.665,1.484-1.484,1.484H1.484C0.665,2.969,0,2.304,0,1.484l0,0C0,0.665,0.665,0,1.484,0 h15.031C17.335,0,18,0.665,18,1.484L18,1.484z"/>
          <path fill="#424242" d="M18,7.516C18,8.335,17.335,9,16.516,9H1.484C0.665,9,0,8.335,0,7.516l0,0c0-0.82,0.665-1.484,1.484-1.484 h15.031C17.335,6.031,18,6.696,18,7.516L18,7.516z"/>
          <path fill="#424242" d="M18,13.516C18,14.335,17.335,15,16.516,15H1.484C0.665,15,0,14.335,0,13.516l0,0 c0-0.82,0.665-1.484,1.484-1.484h15.031C17.335,12.031,18,12.696,18,13.516L18,13.516z"/>
        </svg>
      </a>

      <div class="trigger">
        
          
        
          
        
          
          <a class="page-link" href="/about/index.html">About</a>
          
        
          
        
          
        
      </div>
    </nav>

  </div>

</header>


    <div class="page-content">
      <div class="wrapper">
        <div class="home">

  <!--<h1 class="page-heading">My Blog Posts, in Reverse Chronological Order</h1>-->
  <p class="rss-subscribe">subscribe <a href="/feed.xml">via RSS</a></p>
 
  <!-- This loops through the paginated posts -->
  <!--
  <ul class="post-list">
    
      <li>
        <span class="post-meta">Oct 17, 2015</span>

        <h2>
          <a class="post-link" href="/2015/10/17/trying-github/">Trying GitHub</a>
        </h2>
      </li>
    
      <li>
        <span class="post-meta">Aug 25, 2014</span>

        <h2>
          <a class="post-link" href="/2014/08/25/bay-bridge-during-an-earthquake/">What is the likelihood I'll be on the Bay Bridge during an earthquake?</a>
        </h2>
      </li>
    
      <li>
        <span class="post-meta">Aug 5, 2014</span>

        <h2>
          <a class="post-link" href="/2014/08/05/getting-sequencing-metadata-from-ncbi-sra/">Getting sequencing metadata from NCBI SRA</a>
        </h2>
      </li>
    
      <li>
        <span class="post-meta">May 5, 2014</span>

        <h2>
          <a class="post-link" href="/2014/05/05/who-spit-in-the-chip-seq/">Who spit in the ChIP-seq?</a>
        </h2>
      </li>
    
      <li>
        <span class="post-meta">Mar 31, 2014</span>

        <h2>
          <a class="post-link" href="/2014/03/31/two-by-two/">Two by Two</a>
        </h2>
      </li>
    
  </ul>
  -->

  
    <h1><a href="/2015/10/17/trying-github/">Trying GitHub</a></h1>
       <p class="author">
         <span class="date">Oct 17, 2015</span>
       </p>
    <div class="content">
      <p>I’m trying to get GitHub to serve my pages. This is a test and I’m past the Balmer peak.</p>

    </div>
    <br>
    <br>
    <hr>
    <hr>
    <hr>
    <hr>
    <hr>
    <br>
    <br>
  
    <h1><a href="/2014/08/25/bay-bridge-during-an-earthquake/">What is the likelihood I'll be on the Bay Bridge during an earthquake?</a></h1>
       <p class="author">
         <span class="date">Aug 25, 2014</span>
       </p>
    <div class="content">
      <p>The earthquake early Sunday morning woke me and most of the Bay Area up wondering if a this was the Big One geologist have been warning about.  This quake on the West Napa fault was only 6.1 according to the <a href="//earthquake.usgs.gov/earthquakes/eventpage/nc72282711#summary">USGS</a> but that is not the only fault in the area.  The big worry is the Hayward fault that runs behind my house, through the UC Berkeley Memorial Stadium, past Oakland, and goes down to Fremont.  This fault is pent up, overdue, and UC Davis geologist John Rundle gives 1:5 odds that it will release a 7.0 quake <a href="http://www.sfexaminer.com/sanfrancisco/seismologists-say-bay-area-is-due-for-major-earthquake-that-could-cripple-region/Content?oid=2172405">within the next few years</a>. This is one reason that the Bay Bridge was replaced in 2013, the old structure was crippled in the 1989 quake and was unlikely to survive another serious shake.  </p>

<p>The new eastern span of the Bay Bridge is very pretty, cost $6.4 Billion, and has been plagued with safety concerns that I’m not qualified to judge.  However, I do cross it to get to work and every so often worry about being on it during a quake.  In order to answer the question of what are the chances I’ll be on it during a quake, I looked at my GPS logs that I’ve collected with the <a href="http://theantimap.com/category/applications/antimap-log/">Antimap</a> iPhone <a href="http://itunes.apple.com/app/antimap-log/id486524181">app</a>. This collects time points at a rate of 30 per second and was designed for sports use but I keep it running whenever I’m traveling. Since early 2014 I’ve been a postdoc at UCSF and have a log of every crossing.</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/08/Screen-Shot-2014-08-25-at-3.22.06-PM.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/08/Screen-Shot-2014-08-25-at-3.22.06-PM.png" alt="Screen Shot 2014-08-25 at 3.22.06 PM" /></a></p>

<p>Using the <a href="https://github.com/tkrajina/gpxpy">gpxpy</a> python module I extracted all tracks from January to now that passed those the points on the map.  These correspond to the exit of the Treasure Island tunnel and the point that the Bay Bridge leaves the ground.</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/08/segments.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/08/segments.png" alt="segments" /></a></p>

<p>So how long is each trip across the new bridge?  The mean time spent is 146 seconds, just over two minutes each way.  In total, I’ve spent 5.53 hours crossing that part of the bridge this year.  The logs span 5352 hours so I spend around 1/1000th of my time there. There is a 99.896% chance that I will NOT be on the eastern span at any given time.  Obviously the solution is to go faster across the bridge.</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/08/time_on_bridge.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/08/time_on_bridge.png" alt="time_on_bridge" /></a></p>

    </div>
    <br>
    <br>
    <hr>
    <hr>
    <hr>
    <hr>
    <hr>
    <br>
    <br>
  
    <h1><a href="/2014/08/05/getting-sequencing-metadata-from-ncbi-sra/">Getting sequencing metadata from NCBI SRA</a></h1>
       <p class="author">
         <span class="date">Aug 5, 2014</span>
       </p>
    <div class="content">
      <p>The GEO page <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52386">http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE52386</a> has a link to the SRA study ftp page which you can use to get all the sequencing reads with wget:</p>

<pre><code> wget -r ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByStudy/sra/SRP/SRP033/SRP033009/
</code></pre>

<p>After that, the fast-dump utility from the sra-toolkit can be used to convert the files to gripped fastq files, ready to be mapped with Bowtie, BWA, or other mappers.  However, the file names are uninformative!  Is SRR1029869.fastq.gz a sample of the brain or liver? Is this embryonic or postnatal?</p>

<p>To get the metadata for a study we can use the R package <a href="http://www.bioconductor.org/packages/release/bioc/html/SRAdb.html">SRAdb</a>  and a bit of simple SQL.</p>

<p>Set up by installing SRAdb with Bioconductor and downloading the SRA metainfo database.  This is a large file &gt; 7 GB so keep this in mind for download times and space on your computer.</p>

<p>[code language=”r”]
source(“http://bioconductor.org/biocLite.R”)
biocLite(“SRAdb”)
library(SRAdb)
sqlfile &lt;- getSRAdbFile()
sra_con &lt;- dbConnect(SQLite(),sqlfile)
[/code]</p>

<p>We can get information about each run by looking in the run table:
[code language=”r”]run &lt;- dbGetQuery(sra_con, “select * from run where run_accession = ‘SRR1029856’”)[/code]</p>

<p>[table id=1 /]</p>

<p>This doesn’t tell us very much but it does contain the crucial field that lets us know what experiment it came from: SRX377662.</p>

<p>Using this information, we can look in the experiment table with:</p>

<p>[code language=”r”]exp &lt;- dbGetQuery(sra_con, “select * from experiment where experiment_accession = ‘SRX377662’”)[/code]</p>

<p>[table id=2 /]</p>

<p>This has the title info which we need to make sense of the data: what tissue and when it was collected.</p>

<p>Now we can use the power of SQL and get all the data at once by joining the run and experiment tables and only getting the rows that match the study ID.</p>

<p>[code language=”r”]runs &lt;- dbGetQuery(sra_con, “select run_accession, title from run JOIN experiment USING (experiment_accession) where experiment.study_accession = ‘SRP033009’”)[/code]</p>

<p>[table id=3 /]</p>

    </div>
    <br>
    <br>
    <hr>
    <hr>
    <hr>
    <hr>
    <hr>
    <br>
    <br>
  
    <h1><a href="/2014/05/05/who-spit-in-the-chip-seq/">Who spit in the ChIP-seq?</a></h1>
       <p class="author">
         <span class="date">May 5, 2014</span>
       </p>
    <div class="content">
      <p>A while back in my previous lab, my officemate performed a ChIP experiment using fruit fly embryos. He was disappointed and surprised to see very little signal and started searching for where the experiment went wrong.  There were 95 million reads from the sequencer but when he looked at the reads that matched melanogaster, there were only 4 million! My friend took a handful of the contaminate reads and searched with BLAST to find that a good chunk came from a human genome.  Somewhere along the line, a whole bunch of human DNA got into his experiment.  Being a careful scientist he wanted to know if he made a mistake and contaminated his data so he gave me the reads and asked if I could figure out where the human DNA came from.</p>

<p>The first question I asked was how much of the contaminating DNA is human.  63.5 million reads matched the human reference genome resulting in information on 701 Mb of the genome and only 0.1% of the genome covered with more than 15 reads.  This level of sequencing is not enough to clearly answer an important follow up question: did one person contaminate the sample or many?  An approximation can be made by using the  mitochondria was sequenced to 1500x (off the charts below).  Looking at the mitochondrial variants, that I’m not revealing for privacy reasons, showed that there was only one haplotype.  This doesn’t really rule more than one person out but it supports a single contaminator.</p>

<h1 id="male-or-female">Male or Female?</h1>

<p>If my colleague contaminated the sample, then there should be Y chromosome DNA and half as much X chromosome DNA compared to the autosomes.  The Y chromosome is very degenerated and full of repetitive DNA so we need to filter out all reads that also map to other chromosomes.  Once those are thrown out, there are 7454 reads that mapped the the 59Mb Y chromosome.  Visual inspection of these remaining reads shows that they often have insertions/deletions in the mapping and are suspect.</p>

<p><img src="http://haplotypewriter.com/assets/coverage.png" alt="coverage" /></p>

<p>The relative coverage of the X chromosome is with same range of the autosomes, suggesting whoever contaminated the sample is  XX and not XY.</p>

<h1 id="not-my-officemate-but-who">Not my officemate but who?</h1>

<p>The next person to process the sample was our extremely proficient technician who is normally beyond suspicion but is female along with the rest of the people who process the sample at our sequencing core.  They all fit the genetic profile so far which meant I needed to dig a bit deeper.  To do this I used the <a href="http://www.hagsc.org/hgdp/">Human Genetic Diversity Project</a>, a collection of polymorphism collected from specific populations across the world.  Each population was genotyped at 600,000 positions and I looked at the polymorphisms in the contaminate to see which population it matched the best.  After I did this by hand, a great tool called <a href="http://genome.sph.umich.edu/wiki/LASER">LASER</a> has been released to make this a one step analysis.  Plotting the Unknown along with the HGDP samples shows that it clusters nicely in the Central/South Asian group, ruling out our tech and maintaining her spotless record.</p>

<p><img src="http://haplotypewriter.com/assets/HGDP_pca-01.png" alt="HGDP_pca-01" /></p>

<p>I could show a whole lot more information about this person but won’t because I don’t know who it is and I don’t have consent.  I guess I could try to do a <a href="http://strangervisions.com/">facial reconstruction</a>. I purposely did not look at any health related SNPs. I was even a bit worried that the ratio of X/Y/autosome could reveal something invasive like <a href="http://en.wikipedia.org/wiki/Klinefelter_syndrome">Klinefelter syndrome</a>. At the end of the day, I really wouldn’t be able to say much about this person’s health risk but I can say quite about about her ancestry and so could anyone else who got their hands on a DNA sample.</p>

    </div>
    <br>
    <br>
    <hr>
    <hr>
    <hr>
    <hr>
    <hr>
    <br>
    <br>
  
    <h1><a href="/2014/03/31/two-by-two/">Two by Two</a></h1>
       <p class="author">
         <span class="date">Mar 31, 2014</span>
       </p>
    <div class="content">
      <p>What makes up the mass of a tree? Everyone learns about photosynthesis multiple times in school and yet when MIT and Harvard undergrads were questioned upon graduation “Where does the mass that makes up a tree come from?”, only a few knew the correct answer<a href="http://www.learner.org/resources/series26.html"> http://www.learner.org/resources/series26.html</a>.  Many said sunlight, soil, minerals, oxygen, and water, which are of course involved but most failed to make the connection that carbon dioxide in the air is transformed into 95% of the tree’s mass. People know that plants need sunlight and air and that they make their own food but don’t make the jump to put these pieces of information together and answer this fundamental question. Given this abysmal track record, it is not surprising that when most people are given facts about mutation rates and nuclear decay they fail to understand that these facts have a fundamental impact of the question, where do species come from? In particular, given a mutation rate and an amount of mutations in a population, how long ago did the population share a common ancestor?</p>

<p>Let’s try a few observed differences: If the mutation rate for an organism with a 3 billion nucleotide genome is 2 mutations per 100 million bases per generation and there are zero differences between two individuals then they are clones.  If there are 60 differences between the genomes of two individuals then they must only be one generation apart.  If there are 120 differences, then they must be two generations apart.  There are a few models that fit this observation: there could be two serial generations apart, grandparent and grandchild, or two parallel generations, they share a common parent.</p>

<p><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/mutations-01.png" alt="mutations-01" /></p>

<p>Following some very basic assumptions we can measure how distantly related two individuals are just from counting the number of differences in their genomes.  The more differences, the more generations between them.  If we are sampling individuals from the same generation then we can rule out serial generations and know that we are measuring how long ago they shared a common ancestor. When the size of a population gets to be very very small, the amount of genetic diversity also becomes very small.  There are just less individuals to carry diversity to begin with.  We see this with crop species all the time and it becomes an important consideration when maintaining the genetic health of a population.  Cultivated bananas are clones and from the 1920s -60s, the primary strain <a href="http://en.wikipedia.org/wiki/Gros_Michel">http://en.wikipedia.org/wiki/Gros_Michel </a>was wiped out by a fungus.  A population bottleneck will decrease the amount of diversity and leave a dramatic signature of the population that will persist for many many generations. This is one reason that the measurements that follow are of effective population size in terms of genetic diversity.  The actual number of individuals is going to be larger, usually much larger.</p>

<p>In 2011 Heng Li and Richard Durbin took a single human genome containing a chromosome from each parent and looked are all the places where there are differences between them.  Parts that have fewer differences had a common ancestor more recently than those with more differences.  By scanning across the genome and by noting that because of recombination, different parts of a genome have a different history, they found that the model of the past that best fit the observed data.  Of course, they tested the model with simulations and found that the method of a pairwise sequentially Markovian coalescent (PSMC) recovered the true simulated history.</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/psmc_simulation.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/psmc_simulation.png" alt="psmc_simulation" /></a></p>

<p>Now looking at several sequenced humans and using this same method they confirmed  the out of Africa bottleneck of European and Asian populations.  In this plot, the present is on the left and goes into the past on the right. At no point does the human population go down to Noah, Wife, three daughter-in-laws (10 chromosomes).</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/humans1.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/humans1.png" alt="humans" /></a></p>

<p>Maybe humans are weird and somehow misbehave in this model.  Let’s look at the crawling creatures and the birds of the air.  Many animals have been sequenced and if the population was reduced to two pairs (4 chromosomes) then maybe they will show this signature. The pig is the classic unclean animal and there should only have been two on board.</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/pig-01.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/pig-01.png" alt="pig-01" /></a></p>

<p>The lowest the effective pig population ever sank is around 250 individuals.  How about birds?  Here is the history of the falcons:</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/falcons-01.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/falcons-01.png" alt="falcons-01" /></a></p>

<p>Nope, only went down to 1,500 ish.</p>

<p>The Chimps, Bonobos, Gorillas, and Orangutans that should have been on the Ark:</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/chimps-01.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/chimps-01.png" alt="chimps-01" /></a></p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/gorillas-01.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/gorillas-01.png" alt="gorillas-01" /></a> <a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/ourangutangs-01.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/ourangutangs-01.png" alt="ourangutangs-01" /></a></p>

<p>Other primates:</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/tibet_macaque-01.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/tibet_macaque-01.png" alt="tibet_macaque-01" /></a></p>

<p>Man’s best friend et al:</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/canids.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/canids.png" alt="canids" /></a></p>

<p>The Panda (hope Noah had enough of the single species of bamboo they ate):</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/pandas-01.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/pandas-01.png" alt="pandas-01" /></a></p>

<p>Lions and Tigers:</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/lions_and_tigers-011.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/lions_and_tigers-011.png" alt="lions_and_tigers-01" /></a></p>

<p>How about the reptiles:</p>

<p><a href="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/alligator.png"><img src="http://www.haplotypewriter.com/blog/wp-content/uploads/2014/03/alligator.png" alt="alligator" /></a></p>

<p>Not a single species underwent a severe bottleneck 5-10 thousand years ago.  Even if the dates were wrong, all these species did not experience a bottleneck at the same time. If the tale of Noah’s Ark were true, it should be etched in the genomes of the animals on the boat and it isn’t.</p>

<p>Sources:</p>

<p>Humans: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3154645/">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3154645/</a>
Great Apes: <a href="http://www.nature.com/nature/journal/vaop/ncurrent/full/nature12228.html?WT.ec_id=NATURE-20130704">http://www.nature.com/nature/journal/vaop/ncurrent/full/nature12228.html?WT.ec_id=NATURE-20130704</a>
Tibetan Macaque <a href="http://mbe.oxfordjournals.org/content/early/2014/03/18/molbev.msu104.full.pdf+html">http://mbe.oxfordjournals.org/content/early/2014/03/18/molbev.msu104.full.pdf+html</a>
Malaysian Macaque:<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3491380/"> http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3491380/</a>
Pandas:<a href="http://www.nature.com/ng/journal/v45/n1/full/ng.2494.html?WT.ec_id=NG-201301"> http://www.nature.com/ng/journal/v45/n1/full/ng.2494.html?WT.ec_id=NG-201301</a>
Pigs:<a href="http://www.ncbi.nlm.nih.gov/pubmed/23151582"> http://www.ncbi.nlm.nih.gov/pubmed/23151582</a>
Lions and Snow Leopards: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3778509/#!po=89.2857">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3778509/#!po=89.2857</a>
canids: <a href="http://www.plosgenetics.org/article/info%3Adoi%2F10.1371%2Fjournal.pgen.1004016">http://www.plosgenetics.org/article/info%3Adoi%2F10.1371%2Fjournal.pgen.1004016</a>
Chickens, Alligators, Anolis: <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3760627/#!po=34.0909">http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3760627/#!po=34.0909</a>
Falcons:<a href="http://www.nature.com/ng/journal/v45/n5/full/ng.2588.html"> http://www.nature.com/ng/journal/v45/n5/full/ng.2588.html</a></p>

    </div>
    <br>
    <br>
    <hr>
    <hr>
    <hr>
    <hr>
    <hr>
    <br>
    <br>
  


  
  <div class="pagination">
    
      <span>&laquo; Prev</span>
    

    
      
        <em>1</em>
      
    
      
        <a href="/page2">2</a>
      
    

    
      <a href="/page2">Next &raquo;</a>
    
  </div>
  

  <script type="text/javascript" 
    src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML">
  </script>

</div>

      </div>
    </div>

    <footer class="site-footer">

  <div class="wrapper">

    <h2 class="footer-heading">Haplotypewriter</h2>

    <div class="footer-col-wrapper">
      <div class="footer-col  footer-col-1">
        <ul class="contact-list">
          <li><a href="mailto:aaron.hardin@ucsf.edu">aaron.hardin@ucsf.edu</a></li>
        </ul>
      </div>

      <div class="footer-col  footer-col-2">
        <ul class="social-media-list">
          
          <li>
            <a href="https://github.com/aihardin">
              <span class="icon  icon--github">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M7.999,0.431c-4.285,0-7.76,3.474-7.76,7.761 c0,3.428,2.223,6.337,5.307,7.363c0.388,0.071,0.53-0.168,0.53-0.374c0-0.184-0.007-0.672-0.01-1.32 c-2.159,0.469-2.614-1.04-2.614-1.04c-0.353-0.896-0.862-1.135-0.862-1.135c-0.705-0.481,0.053-0.472,0.053-0.472 c0.779,0.055,1.189,0.8,1.189,0.8c0.692,1.186,1.816,0.843,2.258,0.645c0.071-0.502,0.271-0.843,0.493-1.037 C4.86,11.425,3.049,10.76,3.049,7.786c0-0.847,0.302-1.54,0.799-2.082C3.768,5.507,3.501,4.718,3.924,3.65 c0,0,0.652-0.209,2.134,0.796C6.677,4.273,7.34,4.187,8,4.184c0.659,0.003,1.323,0.089,1.943,0.261 c1.482-1.004,2.132-0.796,2.132-0.796c0.423,1.068,0.157,1.857,0.077,2.054c0.497,0.542,0.798,1.235,0.798,2.082 c0,2.981-1.814,3.637-3.543,3.829c0.279,0.24,0.527,0.713,0.527,1.437c0,1.037-0.01,1.874-0.01,2.129 c0,0.208,0.14,0.449,0.534,0.373c3.081-1.028,5.302-3.935,5.302-7.362C15.76,3.906,12.285,0.431,7.999,0.431z"/>
                </svg>
              </span>

              <span class="username">aihardin</span>
            </a>
          </li>
          

          
          <li>
            <a href="https://twitter.com/aihardin">
              <span class="icon  icon--twitter">
                <svg viewBox="0 0 16 16">
                  <path fill="#828282" d="M15.969,3.058c-0.586,0.26-1.217,0.436-1.878,0.515c0.675-0.405,1.194-1.045,1.438-1.809
                  c-0.632,0.375-1.332,0.647-2.076,0.793c-0.596-0.636-1.446-1.033-2.387-1.033c-1.806,0-3.27,1.464-3.27,3.27 c0,0.256,0.029,0.506,0.085,0.745C5.163,5.404,2.753,4.102,1.14,2.124C0.859,2.607,0.698,3.168,0.698,3.767 c0,1.134,0.577,2.135,1.455,2.722C1.616,6.472,1.112,6.325,0.671,6.08c0,0.014,0,0.027,0,0.041c0,1.584,1.127,2.906,2.623,3.206 C3.02,9.402,2.731,9.442,2.433,9.442c-0.211,0-0.416-0.021-0.615-0.059c0.416,1.299,1.624,2.245,3.055,2.271 c-1.119,0.877-2.529,1.4-4.061,1.4c-0.264,0-0.524-0.015-0.78-0.046c1.447,0.928,3.166,1.469,5.013,1.469 c6.015,0,9.304-4.983,9.304-9.304c0-0.142-0.003-0.283-0.009-0.423C14.976,4.29,15.531,3.714,15.969,3.058z"/>
                </svg>
              </span>

              <span class="username">aihardin</span>
            </a>
          </li>
          
        </ul>
      </div>

      <div class="footer-col  footer-col-3">
        <p class="text">A tangled bank of biology, computers, and Star Wars. More at aaronhardin.com
</p>
      </div>
    </div>

  </div>

</footer>


  </body>

</html>
